sky/compositor/picture_layer.cc

```diff
@@ -0,0 +1,118 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/compositor/picture_layer.h"
+#include "base/logging.h"
+#include "third_party/skia/include/gpu/GrContext.h"
+#include "third_party/skia/include/core/SkSurface.h"
+
+namespace sky {
+namespace compositor {
+PictureLayer::PictureLayer() : last_picture_id_(UINT32_MAX) {
+}
+
+PictureLayer::~PictureLayer() {
+}
+
+void PictureLayer::set_offset(const SkPoint& offset) {
+  offset_ = offset;
+}
+
+void PictureLayer::set_picture(PassRefPtr<SkPicture> picture) {
+  picture_ = picture;
+}
+
+SkMatrix PictureLayer::model_view_matrix(const SkMatrix& model_matrix) const {
+  SkMatrix modelView = model_matrix;
+  modelView.postTranslate(offset_.x(), offset_.y());
+  return modelView;
+}
+
+static void ImageReleaseProc(SkImage::ReleaseContext texture) {
+  DCHECK(texture);
+  reinterpret_cast<GrTexture*>(texture)->unref();
+}
+
+static RefPtr<SkImage> ImageFromPicture(GrContext* context,
+                                        SkPicture* picture,
+                                        const SkRect& paintBounds) {
+  // Step 1: Create a texture from the context's texture provider
+
+  GrSurfaceDesc desc;
+  desc.fWidth = paintBounds.width();
+  desc.fHeight = paintBounds.height();
+  desc.fFlags = kRenderTarget_GrSurfaceFlag;
+  desc.fConfig = kRGBA_8888_GrPixelConfig;
+
+  auto texture = context->textureProvider()->createTexture(desc, true);
```

Comment:
Can you put the type in here so we can see whether its a raw pointer or a smart pointer?